tests/unit/serve/executors/test_executor.py

```diff
@@ -331,3 +338,35 @@ def test_serve(served_exec):
     docs = Client(port=PORT).post(on='/foo', inputs=DocumentArray.empty(5))
 
     assert docs.texts == ['foo' for _ in docs]
+
+
+def test_set_workspace(tmpdir):
+    complete_workspace = os.path.abspath(os.path.join(tmpdir, 'WorkspaceExec', '0'))
+    with Flow().add(uses=WorkspaceExec, workspace=str(tmpdir)) as f:
+        resp = f.post(on='/foo', inputs=Document())
+    assert resp[0].text == complete_workspace
+    with Flow().add(uses=WorkspaceExec, uses_metas={'workspace': str(tmpdir)}) as f:
+        resp = f.post(on='/foo', inputs=Document())
+    assert resp[0].text == complete_workspace
+
+
+def test_default_workspace():
+    with Flow().add(uses=WorkspaceExec) as f:
+        resp = f.post(on='/foo', inputs=Document())
+    assert resp[0].text
+
+    result_workspace = resp[0].text
+    remove_test_dirs(result_workspace)  # delete created directories
```

Comment:
Here I can't use `tmpdir`, since the Executor creates a default workspace, so I need to do cleanup manually.
But this makes me think, do we want to create a dir every time we use and Executor? having a default `workspace` would do that

Reply:
i would avoid creating it if not required.

And for the tmpdir, u can use tmpdir and then u do not need to remove it. It is very dangerous in tests to add remove folder instructions